blockchain/src/genesis.rs

```diff
@@ -36,9 +34,19 @@ impl cbor::CborValue for Body {
         unimplemented!()
     }
     fn decode(value: cbor::Value) -> cbor::Result<Self> {
-        value.iarray().and_then(|array| {
-            Ok(Body { slot_leaders: array })
-        }).embed("While decoding genesis::Body")
+        Ok(Body {
+            slot_leaders: value
+                .iarray()
+                .and_then(|array| {
+                    array.iter().fold(Ok(vec![]), |acc, item| {
+                        acc.and_then(|mut vec| {
```

Comment:
I think this is a bit far fetch there. The fold on an error here is not necessary and is making the code a bit complex to read.

There is an instance of `impl<T> CborValue for LinkedList<T> where T: CborValue`. I agree it is not great to use LinkedList (poor perf and co), but the easy thing to do is:

#### keep the LinkedList in the code

```rust
pub struct Body {
  pub slot_leaders: LinkedList<address::StakeholderId>
}
```

#### leverage the instance of linkedlist

```rust
fn decode(value: cbor::Value) -> cbor::Result<Self> {
    let slot_leaders = cbor::CborValue::decode(value).embed("...")?;
    Ok(Body { slot_leaders })
}
```

This will do all the necessary work for you.